Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py

```diff
@@ -0,0 +1,223 @@
+                        "name",
+                        "slug",
+                        "description",
+                        "groups",
+                        "beta",
+                        "status",
+                        "hr_status",
+                    ],
+                )
+                for a in adapters
+            ],
+        },
+    )
+
+
+def get_connections(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    page = args.get("page", 1)
+    count, connections = octox.get_connections(page=page)
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Connections",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    obj=c,
+                    keys=[
+                        "id",
+                        "adapter_id",
+                        "adapter_name",
+                        "name",
+                        "status",
+                        "description",
+                        "enabled",
+                    ],
+                )
+                for c in connections
+            ],
+        },
+    )
+
+
+def get_discoveries(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    page = args.get("page", 1)
+    count, discoveries = octox.get_discoveries(page=page)
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Discoveries",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    d,
+                    keys=[
+                        "id",
+                        "start_time",
+                        "end_time",
+                        "status",
+                        "hr_status",
+                        "progress",
+                    ],
+                )
+                for d in discoveries
+            ],
+        },
+    )
+
+
+def get_last_discovery(octox: OctoxLabs, *_, **__) -> CommandResults:
+    discovery = octox.get_last_discovery()
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Discovery",
+        outputs=convert_to_json(
+            obj=discovery,
+            keys=["id", "start_time", "end_time", "status", "hr_status", "progress"],
+        ),
+    )
+
+
+def search_devices(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    fields = args.get("fields", None)
+    if isinstance(fields, str):
+        fields = [f.strip() for f in fields.split(",")]
+
+    count, devices = octox.search_assets(
+        query=args.get("query", ""),
+        fields=fields,
+        page=args.get("page", 1),
+        size=args.get("size", 50),
+        discovery_id=args.get("discovery_id", None),
+    )
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Devices",
+        outputs={
+            "count": count,
+            "results": devices,
+        },
+    )
+
+
+def get_device(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    device = octox.get_asset_detail(
```

Comment:
what happens if no results are found?

Reply:
<img width="586" alt="image" src="https://user-images.githubusercontent.com/6793206/190366702-9b03bda0-b428-43dd-969a-150af7cbfa92.png">
Return error message like above image.

Reply:
Let's replace the error with a human readable message telling the users no results were found for the given query

Reply:
Not return as error? Or normal text?